app/lib/features/chat/models/chat_input_state/chat_input_state.dart

```diff
@@ -1,24 +1,28 @@
-import 'package:flutter/material.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
 
 part 'chat_input_state.freezed.dart';
 
+enum SelectedMessageState {
+  none,
+  actions,
+  replyTo,
+  edit,
+}
+
+enum SendingState {
+  preparing,
+  sending,
```

Comment:
Do we not require other states like `sendingFail`, `sent`?

Reply:
No worries, we can keep it simple for now and add those scenario in future based on the requirements.